src/components/base/OakIcon/OakIcon.tsx

```diff
@@ -1,68 +1,24 @@
 import React from "react";
 
-import * as imageMap from "../../../image-map.json";
+import { IconName, icons } from "../../../image-map";
 
 import { OakImage, OakImageProps } from "@/components/base/OakImage";
 
-export const oakIconNames = [
-  "home",
-  "send",
-  "rocket",
-  "video",
-  "edit",
-  "hamburger",
-  "cross",
-  "bell",
-  "twitter",
-  "worksheet",
-  "facebook",
-  "share",
-  "arrow-right",
-  "worksheet-3",
-  "chevron-right",
-  "save",
-  "quiz-3",
-  "chevron-down",
-  "linkedin",
-  "magic-carpet",
-  "books",
-  "supervision-level",
-  "quiz-white",
-  "arrow-left",
-  "additional-material",
-  "slide-deck-3",
-  "sign-language",
-  "external",
-  "equipment-required",
-  "chevron-left",
-  "download",
-  "search",
-  "chevron-up",
-  "go",
-  "copyright",
-  "arrow-down",
-  "project",
-  "quiz",
-  "slide-deck",
-  "content-guidance",
-  "tick",
-  "instagram",
-  "dot",
-  "dot-png",
-  "warning",
-  "lightbulb",
-  "lightbulb-yellow",
-  "intro",
-] as const;
+export const oakIconNames = Object.keys(icons) as IconName[];
 
-export type OakIconName = (typeof oakIconNames)[number];
+export type OakIconName = IconName;
 
 export type OakIconProps = Omit<OakImageProps, "alt" | "src"> & {
   iconName: OakIconName;
   alt?: string;
 };
 
-const typedImageMap = imageMap as Record<string, Record<OakIconName, string>>;
+/**
+ * returns true if the given string is a valid `OakIconName`
+ */
+export function isValidIconName(iconName: string): iconName is OakIconName {
```

Comment:
This will let us narrow an arbitrary string to a bonafide icon name. Needed to pick the correct subject icon for a `subjectSlug`